File: code/java/findDuplicates.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        Integer[] res = new Integer[nums.length];        
        for(int i=0;i<nums.length;i++)
            if(res[nums[i]-1] == null)
                res[nums[i]-1] = -1;
            else
                res[nums[i]-1] ++;
        int index = 0;
        for(int i=0;i<res.length;i++)
            if(res[i] != null)
                if(res[i] == 0)
                    res[index++] = i + 1;
        
        return Arrays.asList(res).subList(0,index);
    }
}

File: code/python/findDuplicates.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        res = [-1 for i in range(len(nums))]
        for i in range(len(nums)):
            res[(nums[i]-1)] += 1
        for i in range(len(res)):
            if res[i] > 0:
                res[i] = i + 1
        index = 0
        while index < len(res):
            if res[index] <= 0:
                res.pop(index)
            else:
                index += 1
        return res
